addon/components/click-outside.js

```diff
@@ -18,7 +18,12 @@ export default Component.extend(ClickOutside, {
       return;
     }
 
-    this.sendAction();
+    let action = this.get('action');
+    if (!action) {
+      return;
+    }
+
+    action(e);
```

Comment:
I think this would be more safe:
```js
if (typeof action !== 'undefined') {
    action(e);
}
```

The `action` callback is a *side-effect*  of the function, not the final result.

Reply:
The function does not return anything, does not prepare a result. It _appends_ to trigger a callback _if it is given_, but this is _optional_. So in that case I try to keep this behavior in the code formatting. But forget it, it's not important.

Reply:
Ah, I see what you mean!